source/ast/expressions/AssertionExpr.cpp

```diff
@@ -1052,8 +1356,13 @@ AssertionExpr& StrongWeakAssertionExpr::fromSyntax(const StrongWeakPropertyExprS
     auto& expr = bind(*syntax.expr, context);
     expr.requireSequence(context);
 
-    if (expr.admitsEmpty())
-        context.addDiag(diag::SeqPropAdmitEmpty, syntax.expr->sourceRange());
+    // Any sequence that is used as a property shall be nondegenerate and shall not admit
+    // any empty match.
+    if (bool isSeqAdmitsEmpty = expr.admitsEmpty(); isSeqAdmitsEmpty || expr.admitsNoMatch()) {
```

Comment:
Rather than checking this again here can you rely on the bind() call above to do it by passing the right flags?

Reply:
currently I don't have any idea how to correctly fix that because there is call to `bind` overload which accepts `SequenceExprSyntax` instead of `PropertyExprSyntax`

there are no way to fix that without changes in `bind` call behavior or `StrongWeakPropertyExprSyntax  struct`